### ISSUE TITLE
Colour space not rendering correctly?

### ISSUE THREAD

**Issue by hogsterbogster** [jurplel/qView#673]
- OS: MacOS 13.6 Ventura
- qView version: 6.1

I'm new to qView, having used the similarly-named JView for years on end ... it was perfect except it didn't anti-alias the images whereas qView does this.

However, qView doesn't seem to render colours correctly. An image in Lightroom looks the same in JView when I export it, but more saturated in qView.

I've tried changing the 'Color space conversion' settings, and Display P3 is the nearest, but it's still not perfect.

Any idea what's going on here?

Please see images below.

Many thanks for your help!

David

Lightroom image:
![lightroom](https://github.com/jurplel/qView/assets/149208922/9a154b23-2196-4499-8b11-ff80b7a22e77)

JView image:
![jview](https://github.com/jurplel/qView/assets/149208922/1b05cb76-5a64-4880-b28c-c4329908ea48)

qView image:
![qview](https://github.com/jurplel/qView/assets/149208922/8598f731-a6ac-4955-836d-44d7d7c84521)

**Comment by pizzabackup** [comment]
same issue for me, images look clearly more saturated than the same file viewed on Mac Preview or dragged in Chrome.

just normal JPG files, srgb profile included.
I've tried all "color space conversion" options also.

**Comment by marjue12255** [comment]
Hello
Yes I have the same problem. Here is an example of a sRGB image on a aRGB monitor. One correctly displayed in Apples Preview App and then in qView. Way too colorful.

MacOS Sonoma 14.4.1
qView Nightly 2024.05.06.1

![SCR-20240510-lwkl](https://github.com/jurplel/qView/assets/96430436/97938c79-9b55-4b96-b289-871250bd5a37)

Greetings
Marcus

[Archiv.zip](https://github.com/jurplel/qView/files/15274386/Archiv.zip)

**Comment by jdpurcell** [comment]
@marjue12255 I was able to reproduce this and it seems to be a bug in Qt ☹️. The image itself worked fine for me, but your monitor's ICC profile has a piece of data that Qt doesn't like. I filed a [bug report](https://bugreports.qt.io/projects/QTBUG/issues/QTBUG-125241) and created a [workaround](https://github.com/jdpurcell/qView/commit/0414d84bd55bb1d192247ef776fece3fddf40fe1) that I could submit as a pull request if @jurplel wants.

As for the original issue reported by @hogsterbogster, it turns out to have a related cause! Lightroom is embedding an ICC profile in the .jpg, which has the same type of data Qt doesn't like, so qView rendered the colors wrong regardless of the monitor. I made sure my workaround is able to correct the issue no matter whether the bug is triggered from the monitor side, image side, or both.
